tests/test_workspaces.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+/// Test adding a second workspace
+#[test]
+fn test_workspaces_add_second_workspace() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "--git", "main"]);
+    let main_path = test_env.env_root().join("main");
+    let secondary_path = test_env.env_root().join("secondary");
+
+    std::fs::write(main_path.join("file"), "contents").unwrap();
+    test_env.jj_cmd_success(&main_path, &["close", "-m", "initial"]);
+
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @"default: 988d8c1dca7e 
+");
+
+    let stdout = test_env.jj_cmd_success(
+        &main_path,
+        &["workspace", "add", "--name", "second", "../secondary"],
+    );
+    insta::assert_snapshot!(stdout.replace('\\', "/"), @r###"
+    Created workspace in "../secondary"
+    Working copy now at: 8ac248e0c8d2 
+    Added 1 files, modified 0 files, removed 0 files
+    "###);
+
+    // Can see the checkout in each workspace in the log output. The "@" node in the
+    // graph indicates the current workspace's checkout.
+    let stdout = test_env.jj_cmd_success(&main_path, &["log", "-T", r#"commit_id " " checkouts"#]);
+    insta::assert_snapshot!(stdout, @r###"
+    o 8ac248e0c8d2d1865fe3679296e329c0137b1a31 second@
+    | @ 988d8c1dca7e0944210ccc33584a6a42cd2962d4 default@
+    |/  
+    o 2062e7d6f1f46b4fe1453040d691931e77a88f7c 
+    o 0000000000000000000000000000000000000000 
+    "###);
+    let stdout = test_env.jj_cmd_success(
+        &secondary_path,
+        &["log", "-T", r#"commit_id " " checkouts"#],
```

Comment:
Nit: can we extract this string/slice so that we have a single one that we more obviously reuse between main_path and secondary_path? Or even as some sort of global (or a function where we pass it the path and it does the right thing), so we can use it in all of the tests in this file?

Reply:
Makes sense, it is quite repetitive. I extracted a function for it.